pkg/kubelet/server/server.go

```diff
@@ -878,6 +890,39 @@ func (s *Server) getPortForward(request *restful.Request, response *restful.Resp
 	proxyStream(response.ResponseWriter, request.Request, url)
 }
 
+// checkpoint handles the checkpoint API request. It checks if the requested
+// podNamespace, pod and container actually exist and only then calls out
+// to the runtime to actually checkpoint the container.
+func (s *Server) checkpoint(request *restful.Request, response *restful.Response) {
```

Comment:
I might have missed it or maybe not part of this PR, when is checkpoint triggered in the container lifecycle? Is it when the container or pod is ready? is it configurable? It would be helpful to trigger the checkpoint if the container is in a certain state.

Reply:
The kubelet API is the only way to trigger a checkpoint, yes.

Reply:
The following question is out of scope for this PR but just picking your brain for future work. 

* Could the trigger to checkpoint container or pod of interest be specified through "container spec" for container checkpoint and "pod spec" for pod checkpoint and at the particular state of the container lifecycle?

Reply:
initially we want to avoid exposing checkpoint operations from the k8s control plane itself.

we need to better understand the pros/cons from a forensic use case.